proto/p4/config/v1/p4types.proto

```diff
@@ -162,6 +163,20 @@ message P4EnumTypeSpec {
   repeated string annotations = 2;
 }
 
+// For serializable (or "unsafe") enums, which have an underlying type. Note
+// that as per the P4_16 specification, the underlying representation can only
+// be a bit<W> type.
+message P4SerializableEnumTypeSpec {
+  message Member {
+    string name = 1;
+    bytes value = 2;
+    repeated string annotations = 3;
```

Comment:
This might apply to other areas of this file as well: in addition to generic `annotations`, should we consider adding an optional `Documentation` field for type definitions? For example, for wither safe or serializable enums I can imagine adding brief and sometimes detailed descriptions of the meaning and usage of each value. Initially they could come from @brief and @description annotations; eventually they could come from javadoc style comments. Again, this is all in the spirit of making the P4Info contain a lot of the needed control plane details.

Reply:
Fair enough!